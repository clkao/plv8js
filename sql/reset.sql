CREATE OR REPLACE FUNCTION test_context_value() RETURNS text as $V8$
    return ctx_value;    
$V8$ LANGUAGE plv8;

CREATE OR REPLACE FUNCTION set_context_value(val TEXT) RETURNS void as $V8$
    ctx_value = val;    
$V8$ LANGUAGE plv8;

SELECT set_context_value('test');
SELECT test_context_value();

SELECT plv8_reset();
SELECT test_context_value();
