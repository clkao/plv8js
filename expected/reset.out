CREATE OR REPLACE FUNCTION test_context_value() RETURNS text as $V8$
    return ctx_value;    
$V8$ LANGUAGE plv8;
CREATE OR REPLACE FUNCTION set_context_value(val TEXT) RETURNS void as $V8$
    ctx_value = val;    
$V8$ LANGUAGE plv8;
SELECT set_context_value('test');
 set_context_value 
-------------------
 
(1 row)

SELECT test_context_value();
 test_context_value 
--------------------
 test
(1 row)

SELECT plv8_reset();
 plv8_reset 
------------
 
(1 row)

SELECT test_context_value();
ERROR:  ReferenceError: ctx_value is not defined
CONTEXT:  test_context_value() LINE 2:     return ctx_value;    
